lib/logger.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * @file Abstracted logging properties and methods.
+ *
+ * If for any reason, they need a different logging tool or need to be extended,
+ * the changes can be made here.
+ */
+var mergeOptions = require('./get-options').mergeOptions;
+var util = require('util');
+var utility = require('./utility');
+var winston = require('winston');
+
+/**
+ * Abstracts fully-fleshed logging module to the logger property of this module.
+ *
+ * If logging with metadata, or requiring winston.log's more advanced features,
+ * use exports.logger().log() instead of exports.log().
+ * @example
+ *   exports.logger(options).log('info', 'Test Log Message', { anything: 'This is metadata' });
```

Comment:
no longer need to inject options from anywhere. `exports.logger` gets the options on it's own